scripts/gha/test_simulator.py

```diff
@@ -499,12 +499,22 @@ def _create_and_boot_simulator(apple_platform, device_name, device_os):
 
   if not device_id:
     # download and create device
-    os.environ["GEM_HOME"] = "$HOME/.gem"
-    args = ["gem", "install", "xcode-install"]
-    logging.info("Download xcode-install: %s", " ".join(args))
+    args = ["brew", "install", "xcodesorg/made/xcodes"]
+    logging.info("Download xcodes: %s", " ".join(args))
     subprocess.run(args=args, check=True)
 
-    args = ["xcversion", "simulators", "--install=%s %s" % (apple_platform, device_os)]
+    # Get the set of available versions for the given Apple platform
+    args = ["xcodes", "runtimes"]
+    runtimes = subprocess.run(args=args, capture_output=True, text=True, check=True)
+    available_versions = re.findall('{0} ([\d|.]+)'.format(apple_platform), runtimes.stdout.strip())
+    logging.info("Found available versions for %s: %s", apple_platform, ", ".join(available_versions))
+
+    # If the requested version is available, use it, otherwise default to the latest
+    if (device_os not in available_versions):
+      logging.warning("Unable to find version %s, will fall back to %s", device_os, available_versions[-1])
```

Comment:
If running in GitHub mode (--ci), print this out in the GitHub warning format to surface it to the overall log.